### PR TITLE
TBD-4991 HDP260 Fix Hive on Spark dependencies

### PR DESCRIPTION
**Pull request by lbourgeois**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

- Test on Spark version is working
- No more several versions of commons-lang

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

**Comment by rdubois** [comment]
I don't understand how this change can fix the TBD-4991, which is more related to a Spark 1.6 issue on:

`Class org.datanucleus.api.jdo.JDOPersistenceManagerFactory was not found.`

The change I see here looks to be for Spark 2.1 even if we remove hive-exec from the Spark 1.6 jobs classpath.

File: main/plugins/org.talend.hadoop.distribution.hdp260/plugin.xml [hunk 1]
```diff
@@ -126,6 +126,13 @@
             name="hive-exec-1.2.1000.2.6.0.3-8.jar" 
             mvn_uri="mvn:org.talend.libraries/hive-exec-1.2.1000.2.6.0.3-8/6.3.0">
         </libraryNeeded>
+
+        <libraryNeeded
+            context="plugin:org.talend.hadoop.distribution.hdp260"
+            id="hive-exec-1.2.1000.2.6.0.3-8-no-commons-lang"
+            name="hive-exec-1.2.1000.2.6.0.3-8-no-commons-lang.jar" 
+            mvn_uri="mvn:org.talend.libraries/hive-exec-1.2.1000.2.6.0.3-8-no-commons-lang/6.0.0">
```
Comment (rdubois) [inline main/plugins/org.talend.hadoop.distribution.hdp260/plugin.xml]: I will need this jar for an upload to nexus.

**Comment by lbourgeois** [comment]
In fact this PR covers to issues :
- Bad component condition update to resolve loading of datanucleus* jars both on Spark1.6 and 2.1
- Multiple versions of commons-lang in classpath (Spark 2.1 only)

File: main/plugins/org.talend.hadoop.distribution.hdp260/plugin.xml (hunk 1 above)
Comment (lbourgeois) [inline main/plugins/org.talend.hadoop.distribution.hdp260/plugin.xml]: I will provide it to you as soon as this PR is merged

**Comment by rdubois** [comment]
Got it. Thanks for the clarification.

**Pull request merged**